### PR TITLE
[PA-5567] : Added platform for amazon-linux-2023 x86_64 & aarch64

### PR DESCRIPTION
**Pull request by span786**
Added platform for amazon-linux-2023 x86_64 & aarch64

### CODE REVIEW

**Comment by span786** [comment]
Agent Runtime builds available [here](https://builds.delivery.puppetlabs.net/puppet-runtime/52513729ae1b805a1512d6d457178ee09cfd724f/artifacts/?C=M&O=D) for main & 7.x branch

File: configs/platforms/amazon-2023-x86_64.rb [hunk 1]
```diff
@@ -0,0 +1,38 @@
+platform "amazon-2023-x86_64" do |plat|
+  plat.servicedir "/usr/lib/systemd/system"
+  plat.defaultdir "/etc/sysconfig"
+  plat.servicetype "systemd"
+
+  packages = %w(
```
Comment (mhashizume) [inline configs/platforms/amazon-2023-x86_64.rb:6]: Why do we inherit from default for aarch64 but not x86_64? It looks like we could remove a lot of redundant packages if we inherited the amazon-2023-x86_64 Vanagon defaults.

Comment (span786) [reply configs/platforms/amazon-2023-x86_64.rb]: I had copied the el-9-x86_64 & el-9-aarch64 platform file for amazon linux 2023. I have now rectified the files for amazon-linux-2023 to have 'inherit_from_default' for both the architectures.

I have also removed the redundant packages and kept only the required ones.

**Comment by span786** [comment]
Agent Runtime builds are available [here](https://builds.delivery.puppetlabs.net/puppet-runtime/34c902af5e2160245c6c50189fd274782d382cc7/artifacts/?C=M&O=D) and their respective Jenkins pipelines build are listed below:

[_agent-runtime-7.x_](https://jenkins-platform.delivery.puppetlabs.net/job/platform_vanagon-generic-builder_vanagon-packaging_generic-builder/2483/console)

1. [amazon-2023-x86_64](https://jenkins-platform.delivery.puppetlabs.net/job/platform_vanagon-generic-builder_vanagon-packaging_generic-builder/BUILD_TARGET=amazon-2023-x86_64,SLAVE_LABEL=k8s-worker/2483/console)
2. [amazon-2023-aarch64](https://jenkins-platform.delivery.puppetlabs.net/job/platform_vanagon-generic-builder_vanagon-packaging_generic-builder/2483/BUILD_TARGET=amazon-2023-aarch64,SLAVE_LABEL=k8s-worker/console)

[_agent-runtime-main_](https://jenkins-platform.delivery.puppetlabs.net/job/platform_vanagon-generic-builder_vanagon-packaging_generic-builder/2484/console) : 

1. [amazon-2023-x86_64](https://jenkins-platform.delivery.puppetlabs.net/job/platform_vanagon-generic-builder_vanagon-packaging_generic-builder/2484/BUILD_TARGET=amazon-2023-x86_64,SLAVE_LABEL=k8s-worker/console)
2. [amazon-2023-aarch64](https://jenkins-platform.delivery.puppetlabs.net/job/platform_vanagon-generic-builder_vanagon-packaging_generic-builder/2484/BUILD_TARGET=amazon-2023-aarch64,SLAVE_LABEL=k8s-worker/console)

File: configs/platforms/amazon-2023-aarch64.rb [hunk 2]
```diff
@@ -0,0 +1,20 @@
+# rubocop:disable Naming/FileName
+# frozen_string_literal: true
```
Comment (mhashizume) [inline configs/platforms/amazon-2023-aarch64.rb:2]: I don't think that you need these Rubocop lines, we don't require Rubocop in this repo:

https://github.com/puppetlabs/puppet-runtime/blob/50ebe6520e0ee418455be5a6fc3ca6dc22163531/Gemfile#L25

**Review by mhashizume — APPROVED** [review]
This looks good, we can circle back to the Rubocop comments later

**Comment by joshcooper** [comment]
Going to merge this as pxp-agent-vanagon is attempting to build on amazon (but we haven't built the runtime yet). @span786 could you submit a maint PR tomorrow to resolve the rubocop stuff?

**Comment by span786** [comment]
Raised maintenance PR (https://github.com/puppetlabs/puppet-runtime/pull/749) to address rubocop stuff

File: configs/platforms/amazon-2023-aarch64.rb (hunk 2 above)
Comment (span786) [reply configs/platforms/amazon-2023-aarch64.rb:2]: Raised maintenance PR (https://github.com/puppetlabs/puppet-runtime/pull/749) to address this

**Pull request merged**